include/class.api.php

```diff
@@ -244,14 +244,17 @@ function getRequestStructure($format, $data=null) { return array(); }
      * expected. It is assumed that the functions actually implementing the
      * API will further validate the contents of the request
      */
-    function validateRequestStructure($data, $structure, $prefix="", $strict=true) {
+    function validateRequestStructure($data, $structure, $prefix="", $strict=true, $level=0) {
         global $ost;
 
         foreach ($data as $key=>$info) {
+            if($level == 6)
+                break;
             if (is_array($structure) && (is_array($info) || $info instanceof ArrayAccess)) {
+                $level++;
                 $search = (isset($structure[$key]) && !is_numeric($key)) ? $key : "*";
                 if (isset($structure[$search])) {
-                    $this->validateRequestStructure($info, $structure[$search], "$prefix$key/", $strict);
+                    $this->validateRequestStructure($info, $structure[$search], "$prefix$key/", $strict, $level);
```

Comment:
You should probable do `$level + 1` in the nested call so that `$level` is not corrupted inside the `foreach` loop. Otherwise, the loop will validate combinations of six items in a loop, including nesting.

Reply:
Thank you, will look into this.